core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -70,4 +73,26 @@ public static String propertyAsString(Map<String, String> properties,
     }
     return defaultValue;
   }
+
+  /**
+   * Returns subset of provided map with keys matching the provided prefix. Matching is case-sensitive and the matching
+   * prefix is removed from the keys in returned map.
+   *
+   * @param properties input map
+   * @param prefix prefix to choose keys from input map
+   * @return subset of input map with keys starting with provided prefix and prefix trimmed out
+   */
+  public static Map<String, String> propertiesWithPrefix(Map<String, String> properties,
```

Comment:
prefer method arguments all on next lines of it has to span multiple lines, e.g. 

```java
propertiesWithPrefix(
  Map<String, String> properties,
  String prefix) {
```